specs/online_order_spec.rb

```diff
@@ -1,90 +1,115 @@
       # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org",
+        {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip=>"98872-9105"})
+
+      online_order = Grocery::OnlineOrder.new(1, products, customer, :complete)
+
+      online_order.customer.must_be_instance_of Grocery::Customer
     end
 
     it "Can access the online order status" do
       # TODO: Your test code here!
-    end
-  end
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      customer = Grocery::Customer.new(1, "leonard.rogahn@hagenes.org",
+        {:street=>"71596 Eden Route", :city=>"Connellymouth", :state=>"LA", :zip=>"98872-9105"})
 
-  describe "#total" do
-    it "Adds a shipping fee" do
-      # TODO: Your test code here!
-    end
+      online_order = Grocery::OnlineOrder.new(1, products, customer, :complete)
 
-    it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      online_order.status.must_equal :complete
     end
   end
 
-  describe "#add_product" do
-    it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
-    end
 
-    it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
-    end
-  end
 
   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
       # TODO: Your test code here!
+      result = Grocery::OnlineOrder.all
+      result.must_be_instance_of Array
     end
 
     it "Returns accurate information about the first online order" do
       # TODO: Your test code here!
+      online_orders = Grocery::OnlineOrder.all
+      first_order = online_orders[0]
+
+       first_order.id.must_equal 1
+       #first_order.products.must_equal "Lobster"=>17.18, "Annatto seed"=>58.38, "Camomile"=>83.21
+        first_order.customer.id.must_equal 25
+       # first_order.status.must_equal :complete
     end
 
     it "Returns accurate information about the last online order" do
       # TODO: Your test code here!
+      Grocery::OnlineOrder.all[-1].id.must_equal 100
+      Grocery::OnlineOrder.all[-1].status.must_equal :pending
     end
   end
-
   describe "OnlineOrder.find" do
     it "Will find an online order from the CSV" do
       # TODO: Your test code here!
+
+      Grocery::OnlineOrder.find(1).id.must_equal 1
+      Grocery::OnlineOrder.find(3).status.must_equal :processing
     end
 
     it "Raises an error for an online order that doesn't exist" do
       # TODO: Your test code here!
+      proc {Grocery::OnlineOrder.find(206)}.must_raise ArgumentError
     end
   end
 
   describe "OnlineOrder.find_by_customer" do
     it "Returns an array of online orders for a specific customer ID" do
       # TODO: Your test code here!
+
+      result = Grocery::OnlineOrder.find_by_customer(1)
+      result.must_be_instance_of Array
```

Comment:
You should also check to see:
1.  If it has the right number of elements
1.  That each element is an `OnlineOrder`
1.  If those elements have the right order ids